content/en/blog/_posts/2022-11-29-validating-admission-policies-alpha/index.md

```diff
@@ -0,0 +1,150 @@
+```
+
+Note that the `expression` field contains a CEL expression that will be
+evaluated by all resources matching the `matchConstraints`.
+
+Next bind the policy the the appropriate resources:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "demo-binding-test.example.com"
+spec:
+  policy: "replicalimit-policy.example.com"
+  matchResources:
+    namespaceSelectors:
+    - key: environment,
+      operator: In,
+      values: ["test"]
+```
+
+This `ValidatingAdmissionPolicyBinding` resource binds the policy we just
+defined only to for namespaces where the `environment` label is set to `test`.
+
+Compared to an admission webhook, admission policies are far simpler. The entire
+policy is declared in Kubernetes resources. Unlike webhooks, there are no
+external binaries or remote requests involved.
+
+Validation admission policies are highly configurable, enabling policy authors
+to define policies that can be parameterized and scoped to resources as needed
+by cluster administrators.
+
+For example, the above admission policy can be modified to make it configurable:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "replicalimit-policy.example.com"
+Spec:
+  paramKind:
+    apiVersion: rules.example.com/v1
+    kind: ReplicaLimit
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  validations:
+    - expression: "object.spec.replicas <= params.maxReplicas"
+```
+
+Here, `paramKind` defines the resources used to configure the policy and the
+`expression` uses the `params` variable to access the parameter resource.
+
+This allows multiple bindings to be defined, each configured differently. For
+example:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "replicalimit-binding-production.example.com"
+spec:
+  policy: "replicalimit-policy.example.com"
+  paramsRef:
+    name: "replica-limit-production.example.com"
+  matchResources:
+    namespaceSelectors:
+    - key: environment,
+      operator: In,
+      values: ["production"]
+```
+
+```yaml
+apiVersion: rules.example.com/v1
+kind: ReplicaLimit
+metadata:
+  name: "replica-limit-production.example.com"
+maxReplicas: 1000
+```
+
+This binding and parameter resource configure deployments in namespaces with the
+`environment` label set to `production` to a max limit of 1000 replicas.
+
+A seperate binding can then be configured to set a lower limit for `test`
+namespaces.
+
+I hope this has given you a glimpse of what is possible with validating
+admission policies! There are many features that we have not yet touched on.
+Have a look at the [feature
+documentation](content/en/docs/reference/access-authn-authz/validating-admission-policy.md)
+to learn more.
```

Comment:
```suggestion
To learn more, read
[FIXME](/docs/reference/access-authn-authz/validating-admission-policy/).
```

Use the title of the page there. However, https://deploy-preview-37810--kubernetes-io-main-staging.netlify.app/docs/reference/access-authn-authz/validating-admission-policy/ doesn't work yet. Is there a PR that needs to land first?

Reply:
Correct, this requires https://github.com/kubernetes/website/pull/37770